test/common.py

```diff
@@ -2395,8 +2389,27 @@ def log_request(code=0, size=0):
   # do not have the correct MIME type
   SimpleHTTPRequestHandler.extensions_map['.mjs'] = 'text/javascript'
 
-  httpd = ThreadingHTTPServer(('localhost', port), TestServerHandler)
-  httpd.serve_forever() # test runner will kill us
+  return ThreadingHTTPServer(('localhost', port), TestServerHandler)
+
+
+class ServerThread(threading.Thread):
+  """A generic thread class to create and run a server."""
+  def __init__(self, server_creator, *server_args):
```

Comment:
`server_ctor` or `server_constructor`.. or maybe `server_class`?

Reply:
It takes a function, so I don't think those names work.